activity_browser/ui/tables/activity.py

```diff
@@ -91,7 +98,28 @@ def get_interpreter(self):
         return self.model.get_interpreter()
 
     def selected_exchanges(self) -> List[any]:
-        return [self.model.get_exchange(index) for index in self.selectedIndexes()]
+        # The context menu still makes sense in read-only mode, but
+        # item selection is disabled, so we need to use the current item
+        if self.selectedIndexes():
+            return [self.model.get_exchange(index) for index in self.selectedIndexes()]
+        else:
+            return [self.model.get_exchange(self.currentIndex())]
+    
+    def set_read_only(self, read_only: bool):
```

Comment:
This name needs to be easier to understand - maybe `set_read_only_flag` or `set_read_only_state`?